rfd/0208-selinux-policy.md

```diff
@@ -0,0 +1,188 @@
+
+The SELinux module will be updated in the future to support new features and maintain compatibility with existing features.
+Any changes to the SSH service that result in new files being managed, new network activity (listening on new ports, dialing new addresses etc),
+new binaries being executed, or new Linux syscalls being called will require updates to the SELinux module.
+
+### Installation and updating
+
+The module source will be embedded in future Teleport agent binaries. This will ensure that the module is always compatible with the installed version of Teleport.
+
+`teleport-update` will be modified to support managing SELinux modules for Teleport SSH. `teleport-update` will install the SELinux
+module if configured to, and will remove it if it was previously installed and the user specified that they don't want the module
+installed now. If SELinux is not present on the host, the `teleport-update` will tell the user this and exit. If SELinux is present but 
+the Teleport agent module is not installed, the `teleport-update` will extract the embedded module and use `checkmodule`, `semodule_package`,
+and `semodule` to compile and install the module. Directories and files that Teleport SSH requires will be created if they do not exist
+after the module is installed, so that they can be labeled correctly. If necessary files do not have the correct SELinux labels Teleport SSH
+will not be able to function properly.
+If `teleport-update` is unable to install the module, it will rollback to the module that was previously installed, if any.
+
+`teleport-update` will only manage the official Teleport SELinux module. Modules created by users or third parties will not be inspected or
+modified. If an SELinux module installed that shares a name with the official Teleport SELinux module it will be assumed to be a version of
+the Teleport SELinux module and may be modified.
+
+If the SSH service itself was allowed to manage its own SELinux module, an attacker that compromised the SSH service could modify the module to
+be more permissive, defeating the point of the module in the first place.
+
+Installation scripts will be updated to install and configure the SELinux module with the `teleport-update`.
+
+SELinux has 3 modes, disabled, permissive (module violations are only logged) and enforcing (attempts to violate the module are denied).
+The SELinux mode will not be modified by `teleport-update`, users will be responsible for enabling SELinux if they so choose.
+
+The `teleport-update` will have a `--dry-run` flag added to the necessary subcommands that will print what actions it would need to take to
+manage the module if run normally and print the module that would be installed based on the provided Teleport agent configuration file.
+
+The following RPM packages will be required to install the SELinux module:
+- selinux-policy-devel
+- policycoreutils
+
+#### Examples
+
+```
+# installing a module
+$ teleport-update enable --enable-selinux
```

Comment:
It's not super clear, does this enable both managed updates and SELinux? How do I only install/update the SELinux policy without enabling auto-updates?

Reply:
Changed the comment to clarify, using `teleport-update` will enable managed updates. Added an example of using the install script.